src/contracts/atlas/GasAccounting.sol

```diff
@@ -85,29 +93,31 @@ abstract contract GasAccounting is SafetyLocks {
         if (_lock.activeEnvironment != environment) {
             revert InvalidExecutionEnvironment(_lock.activeEnvironment);
         }
-        if (_lock.phase != uint8(ExecutionPhase.SolverOperations)) {
+        if (_lock.phase != uint8(ExecutionPhase.SolverOperation)) {
             revert WrongPhase();
         }
 
         (address currentSolver, bool calledBack, bool fulfilled) = _solverLockData();
 
-        if (calledBack) revert DoubleReconcile();
-
         if (solverFrom != currentSolver) revert InvalidSolverFrom(currentSolver);
```

Comment:
Similar question as above about passing in solverFrom as a param. We read it from storage anyway, has to be the current solver, and it's trivial to read the current solver from Atlas by calling `solver()` if someone was calling this maliciously.

So if no protection, maybe we can remove as a param?